src/dune_engine/dep.ml

```diff
@@ -77,15 +77,15 @@ module Fact = struct
     type t =
       { files : Digest.t Path.Map.t
       ; dirs : Digest.t Path.Map.t (* Only for file selectors for now *)
-      ; parent_dirs : Path.Set.t
+      ; dirs_that_must_exist : Path.Build.Set.t
```

Comment:
@snowleopard btw, should this be a field at all? It seems like something that can be computed on the fly or at least be lazily computed?

Reply:
Indeed, it seems weird to keep it in memory all the time. Let's drop it!

Reply:
Turns that we do use the same value more than once and this has the potential to lower the amount of `mkdir` calls. I don't have the bandwidth to see if this is worth it now, but I left a comment for further investigation if it's worth making this lazy.